src/DotVVM.Framework/Diagnostics/CompilationPageViewModel.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using DotVVM.Framework.Compilation;
+using DotVVM.Framework.ViewModel;
+
+namespace DotVVM.Framework.Diagnostics
+{
+    public class CompilationPageViewModel : DotvvmViewModelBase
+    {
+        private readonly IDotvvmViewCompilationService viewCompilationService;
+
+        public ImmutableArray<DotHtmlFileInfo> Routes => viewCompilationService.GetRoutes();
+        public ImmutableArray<DotHtmlFileInfo> MasterPages => viewCompilationService.GetMasterPages();
+        public ImmutableArray<DotHtmlFileInfo> Controls => viewCompilationService.GetControls();
+        public int ActiveTab { get; set; } = 0;
+
+        public CompilationPageViewModel(IDotvvmViewCompilationService viewCompilationService)
+        {
+            this.viewCompilationService = viewCompilationService;
+        }
+
+        public override async Task Init()
+        {
+            var isAuthorized = await Context.Configuration.Diagnostics.CompilationPage.AuthorizationPredicate(Context);
+            if (!isAuthorized)
+            {
+                Context.HttpContext.Response.StatusCode = 403;
+                Context.InterruptRequest();
+                return;
+            }
+
+            await base.Init();
+        }
+
+        public async Task CompileAll()
+        {
+            await viewCompilationService.CompileAll(true, true);
```

Comment:
```suggestion
            await viewCompilationService.CompileAll(buildInParallel: true, forceRecompile: true);
```

And I don't think forcing recompilation is good default. Could we have a switch or second button for this in the UI if it's needed in some cases.